include/sparrow/fixed_size_layout.hpp

```diff
@@ -218,33 +219,33 @@ namespace sparrow
      ***********************************/
 
     template <class T>
-    fixed_size_layout<T>::fixed_size_layout(array_data ad)
-        : m_data(std::move(ad))
+    fixed_size_layout<T>::fixed_size_layout(array_data& data)
+        : p_data(&data)
     {
         // We only require the presence of the bitmap and the first buffer.
-        assert(m_data.buffers.size() > 0);
-        assert(m_data.length == m_data.bitmap.size());
+        assert(p_data->buffers.size() > 0);
```

Comment:
We might need a free function checking if an `array_data` seems well-formed, and use that in asserts too.

Reply:
Whether an `array_data` is well-formed is actually specific to the layout (even if some layouts share some invariants).

Reply:
In that case it might be necessary to make a concept for layouts that requests, among other thing, that these layout types provide each a way to check if the provided `array_data` matches the expected structure.
Then, because of the nature of the construction of the layout, the check must be done whatever the build context and then we must throw an exception (or abort, that depends on what we decide to support as error handling) or keep assertions and *document*  that providing an `array_data` to constructor layout that doesnt match the expected structure is undefined (and maybe provide a way to activate the check).
Lots of tradeoff choices here.

Reply:
I think we should do this in a dedicated PR. I'll open an issue to track it.